controller.json

```diff
@@ -0,0 +1,564 @@
+{
+	"$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+	"contentVersion": "0.0.0.0",
+	"parameters": {
+		"identifier": {
+			"type": "string",
+			"minLength": 1,
+			"maxLength": 16,
```

Comment:
Why limit this to 16 characters?

Reply:
The identifier gets appended to other strings to form resource names and those resource names have varying length limits. 16 seemed both long enough to offer an easily identifiable name and short enough to still be safe when appended to other strings and used as resource names. We can probably make it longer than 16 if you think a higher limit is worthwhile.

Reply:
No that's good enough justification for me!